pkg/controller/controller_instance.go

```diff
@@ -128,6 +129,46 @@ func (c *controller) reconcileServiceInstanceDelete(instance *v1alpha1.ServiceIn
 		return nil
 	}
 
+	// Determine if any bindings exist for this instance
+	bindingsLister := c.bindingLister.ServiceInstanceCredentials(instance.Namespace)
```

Comment:
Nit: it's probably better to use 'instance credentials' instead of 'bindings' at this point.